### PR TITLE
Add Elasticwriter feature

### PR DESCRIPTION
**Pull request by Crunsher**
Refs #5538

In a  working but unfinished state

### CODE REVIEW

File: lib/perfdata/elasticwriter.cpp [hunk 1]
```diff
@@ -0,0 +1,320 @@
+/******************************************************************************
```
Comment (gunnarbeutner) [inline lib/perfdata/elasticwriter.cpp]: Something went wrong here. :P

**Comment by dnsmichi** [comment]
The implementation review happened directly in #5538 and we've worked offline on the final commit.

**Comment by lazyfrosch** [comment]
What about authentication against Elasticsearch?

**Comment by dnsmichi** [comment]
Note: This feature will be renamed for 2.8 to better reflect its purpose. `ElasticsearchWriter` specifies the backend, no need for a full Elastic stack here.

**Pull request merged**